### PR TITLE
feat: add get username from uuid

### LINKED ISSUE #6: Method to get player's name from UUID
### Feature

When writing user facing applications it is annoying to have to externally fetch the user's name from an external source.

### Ideal solution or implementation

```ts
const username = await client.util.getUsername("UUID")
```

### Alternative solutions or implementations

Using axios and writing it myself :-(

### Other context

_No response_

### PR DESCRIPTION
**Pull request by chriscn**
Closes my own issue lol closes #6

### CODE REVIEW

File: src/classes/Util.ts [hunk 1]
```diff
@@ -39,4 +39,20 @@ export class Util extends BaseClass {
 		const regex = /^[0-9a-f]{32}$/i;
 		return regex.test(uuid);
 	}
+
+	/**
+	 * Get player's name from their UUID
```
Comment (FC5570) [inline src/classes/Util.ts:44]: Please also add comments for the uuid parameter.

File: src/classes/Util.ts [hunk 2]
```diff
@@ -39,4 +39,20 @@ export class Util extends BaseClass {
...
+	 */
+	public async getUsername(uuid: string) {
+		try {
+			const data = await petitio(`https://api.mojang.com/user/profile/${uuid}`).send();
+			const json = (await data.json()) as GetUsernameResponse;
+
+			if (json.error) throw new HypixelTSError('GET_USERNAME_ERROR', json.error, data.statusCode);
```
Comment (FC5570) [inline src/classes/Util.ts:51]: You need to add the error message, an error will be thrown with the current implementation, check this file and add the messages: https://github.com/hypixelts/hypixel.ts/blob/3890323571ca8be3a99c013b067476edddf0357c/src/errors/Messages.ts

File: src/classes/Util.ts [hunk 3]
```diff
@@ -39,4 +39,20 @@ export class Util extends BaseClass {
...
+
+			return json.name;
+		} catch {
+			throw new HypixelTSError('GET_USERNAME_404');
```
Comment (FC5570) [inline src/classes/Util.ts:55]: Need to add error message for this key too.

**Review by FC5570 — CHANGES_REQUESTED** [review]
Apart from these. Please also:

- Add tests
- Place the `getUsername` function below the `getUUID` function.

**Comment by chriscn** [comment]
Only failing test is now
`FAIL  tests/classes/Player.test.ts > Player.recentlyPlayedGames returns Array of recently played games
Error: Test timed out in 5000ms.` which is unrelated.

File: src/classes/Util.ts (hunk 1 above)
Comment (chriscn) [reply src/classes/Util.ts:34]: fixed in 3dabc588

File: src/classes/Util.ts (hunk 2 above)
Comment (chriscn) [reply src/classes/Util.ts:43]: fixed in a7c0b62e

**Review by chriscn — COMMENTED** [review]
Hardcoding the regex doesn't make any sense in this test. If the backend function changed the test would still pass.

**Comment by chriscn** [comment]
If you merge this, I would suggest the `rebase` option so that commit history is preserved

**Comment by FC5570** [comment]
Closed in favor of #11

**Pull request closed without merging**